src/Video/Renderer.cpp

```diff
@@ -66,84 +59,71 @@ Renderer::Renderer() {
-    lighting->ClearLights();
-    
-    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    renderSurface->Clear();
     glViewport(0, 0, static_cast<GLsizei>(renderSurface->GetSize().x), static_cast<GLsizei>(renderSurface->GetSize().y));
 }
 
-void Renderer::AddLight(const Video::Light& light) {
-    lighting->AddLight(light);
-}
+void Renderer::SetLights(const std::vector<Video::Light>& lights) {
+    lightCount = lights.size();
 
-void Renderer::Light(const glm::mat4& inverseProjectionMatrix, RenderSurface* renderSurface) {
-    lighting->Render(inverseProjectionMatrix, renderSurface);
+    // Skip if no lights.
+    if (lightCount == 0) return;
+
+    // Resize light buffer if necessary.
+    unsigned int byteSize = sizeof(Video::Light) * lights.size();
+    if (lightBuffer->GetSize() < byteSize) {
+        delete lightBuffer;
+        lightBuffer = new StorageBuffer(byteSize, GL_DYNAMIC_DRAW);
+    }
+
+    // Update light buffer.
+    lightBuffer->Bind();
+    lightBuffer->Write((void*)lights.data(), 0, byteSize);
+    lightBuffer->Unbind();
 }
 
 void Renderer::PrepareStaticMeshRendering(const glm::mat4& viewMatrix, const glm::mat4& projectionMatrix) {
-    staticRenderProgram->PreRender(viewMatrix, projectionMatrix);
+    staticRenderProgram->PreRender(viewMatrix, projectionMatrix, lightBuffer, lightCount);
 }
 
 void Renderer::RenderStaticMesh(Geometry::Geometry3D* geometry, const Texture2D* albedo, const Texture2D* normal, const Texture2D* metallic, const Texture2D* roughness, const glm::mat4 modelMatrix, bool isSelected) {
     staticRenderProgram->Render(geometry, albedo, normal, metallic, roughness, modelMatrix, isSelected);
 }
 
 void Renderer::AntiAlias(RenderSurface* renderSurface) {
-    //fxaaFilter->SetScreenSize(renderSurface->GetSize());
-    //postProcessing->ApplyFilter(renderSurface, fxaaFilter);
-}
-
-void Renderer::RenderFog(RenderSurface* renderSurface, const glm::mat4& projectionMatrix, float density, const glm::vec3& color) {
-    /*fogFilter->SetProjectionMatrix(projectionMatrix);
-    fogFilter->SetDensity(density);
-    fogFilter->SetColor(color);
-    postProcessing->ApplyFilter(renderSurface, fogFilter);*/
-}
-
-void Renderer::ApplyGlow(RenderSurface* renderSurface, int blurAmount) {
-    /*glowBlurFilter->SetScreenSize(renderSurface->GetSize());
-    for (int i = 0; i < blurAmount; ++i) {
-        glowBlurFilter->SetHorizontal(true);
-        postProcessing->ApplyFilter(renderSurface, glowBlurFilter);
-        glowBlurFilter->SetHorizontal(false);
-        postProcessing->ApplyFilter(renderSurface, glowBlurFilter);
-    }
-    postProcessing->ApplyFilter(renderSurface, glowFilter);*/
-}
-
-void Renderer::ApplyColorFilter(RenderSurface* renderSurface, const glm::vec3& color) {
-    /*colorFilter->SetColor(color);
-    postProcessing->ApplyFilter(renderSurface, colorFilter);*/
+    fxaaFilter->SetScreenSize(renderSurface->GetSize());
+    postProcessing->ApplyFilter(renderSurface, fxaaFilter);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'AntiAlias' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)